File: src/main/java/ast/ASTClassDeclStmnt.java
package ast;

public class ASTClassDeclStmnt extends ASTNode {
    public enum Type {
        CLASS, OBJECT, STRUCT
    }
    public ASTNodeArray<ASTVarDecl> fields;
    public ASTNodeArray<ASTFuncDecl> methods;
    public Type type;
    public String superclass;
    public String name;

    public ASTClassDeclStmnt(ASTFileLocation loc, String name, ASTNodeArray<ASTVarDecl> fields, ASTNodeArray<ASTFuncDecl> methods, Type type, String superclass){
        super(loc);
        this.name = name;
        this.fields = fields;
        this.methods = methods;
        this.type = type;
        this.superclass = superclass;
    }

    public void print(int i) {
        printIndent(i);
        System.out.println("ASTClassDeclStmnt: (name, superclass, type, fields, methods)");
        printIndent(i+1);
        System.out.println(name);
        printIndent(i+1);
        System.out.println(superclass);
        printIndent(i+1);
        System.out.println(type);
        fields.print(i+1);
        methods.print(i+1);
    }

}

File: src/main/kotlin/ast/ASTClassDeclStmnt.kt
package ast

import compiler.*

class ASTClassDeclStmnt(loc: ASTFileLocation, val name: String, val fields: ASTNodeArray<ASTVarDecl>, val methods: ASTNodeArray<ASTFuncDecl>, val type: Type, val superclass: String?) : ASTNode(loc) {

    val scope = SymbolTable(null)
    fun setParentScope(pscope: SymbolTable?) {
        scope.parent = pscope
    }

    enum class Type {
        CLASS, OBJECT, STRUCT
    }

    override fun print(i: Int) {
        printIndent(i)
        println("ASTClassDeclStmnt: (name, superclass, type, fields, methods)")
        printIndent(i + 1)
        println(name)
        printIndent(i + 1)
        println(superclass)
        printIndent(i + 1)
        println(type)
        fields.print(i + 1)
        methods.print(i + 1)
    }

}